client/js/lib/segment/useSegmentNavigation.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * (c) 2010-present DEMOS plan GmbH.
+ *
+ * This file is part of the package demosplan,
+ * for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+import { checkResponse, dpApi } from '@demos-europe/demosplan-ui'
+
+/**
+ * Validates if a string is a valid UUID v4 format
+ * @param {string} uuid - The UUID string to validate
+ * @returns {boolean} True if valid UUID format
+ */
+export function isValidUUID (uuid) {
+  const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+  return uuidRegex.test(uuid)
+}
+
+/**
+ * Composable for segment navigation with pagination
+ * Handles URL-based navigation to specific segments with automatic pagination
+ *
+ * @param {string} statementId - The parent statement ID
+ * @param {string} storageKey - LocalStorage key for pagination persistence
+ * @param {Object} paginationRef - Reactive pagination object reference
+ * @param {Object} defaultPagination - Default pagination settings
+ * @returns {Object} Methods for segment navigation
+ */
+export function useSegmentNavigation (statementId, storageKey, paginationRef, defaultPagination) {
+  /**
+   * Fetches the position of a segment within its statement
+   * @param {string} segmentId - The segment ID to get position for
+   * @returns {Promise<Object|null>} Position data or null on error
+   */
+  async function getSegmentPosition (segmentId) {
+    // Validate UUID format before making API call
+    if (!isValidUUID(segmentId)) {
+      console.error('Invalid segment ID format:', segmentId)
+      return null
+    }
+
+    try {
+      const url = Routing.generate('dplan_segment_position', {
+        segmentId: segmentId,
+        statementId: statementId
+      })
+
+      return await dpApi.get(url)
+        .then(checkResponse)
+    } catch (error) {
+      console.error('Failed to get segment position:', error)
+      return null
+    }
+  }
+
+  /**
+   * Calculates the correct page for a segment and updates pagination
+   * Should be called once during component initialization if segment param exists
+   *
+   * @param {Object} context - Context object
```

Comment:
Why the `context` object, if we only need `hasNavigatedToSegment`? We could just pass that as a parameter directly